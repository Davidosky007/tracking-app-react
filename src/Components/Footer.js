import { logout } from '../logic/sessions';
import Icon from './Icon';

const Footer = () => {
  const handleLogout = () => {
    logout();
  };

  return (
    <footer className="flex fixed footer">
      <Icon icon="fas fa-plus-square" path="/add" title="Add Measure." />
      <Icon icon="fas fa-home" path="/" title="Home" />
      <Icon icon="fas fa-weight" path="/progress" title="Progress" />
      <Icon icon="fas fa-sign-out-alt" path="/users/login" title="Logout" clickHandler={handleLogout} />
    </footer>
  );
};

export default Footer;
