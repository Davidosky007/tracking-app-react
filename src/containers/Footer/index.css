.footer {
  width: 100%;
  justify-content: space-between !important;
}

.footer :focus {
  background-color: #62b5e5;
}
