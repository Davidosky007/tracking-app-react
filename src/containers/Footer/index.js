import React from "react";
import { useLocation } from 'react-router-dom';
import "./index.css";

import { BottomNavigationAction, BottomNavigation } from "@material-ui/core";
import { FontAwesomeIcon } from "@fortawesome/react-fontawesome";
import {
  faChartLine,
  faChartPie,
  faEllipsisH,
  faChartBar,
} from "@fortawesome/free-solid-svg-icons";
import { useHistory } from "react-router";

const Footer = () => {
  const history = useHistory();
  const [value, setValue] = React.useState("addmeasure");

  const handleChange = (newValue) => {
    setValue(newValue);
  };

  const location = useLocation();
  const path = location.pathname.split('/')[2];

  const signUp = path !== 'signin';
  const validatePassword = signUp;

  return (
    <BottomNavigation
      showLabels
      value={value}
      onChange={handleChange}
      className="footer bg-dark h-100 w-100"
    >
      <BottomNavigationAction
        className="p-1 text-white"
        label="Add Measure"
        value="addmeasure"
        onClick={() => {
          history.push("/new");
        }}
        icon={<FontAwesomeIcon className="m-1 text-white" icon={faChartBar} />}
      ></BottomNavigationAction>
      <BottomNavigationAction
        className="p-1 text-white"
        label="Track"
        value="trackit"
        onClick={() => {
          history.push("/measurements");
        }}
        icon={<FontAwesomeIcon className="m-1 text-white" icon={faChartLine} />}
      />
      <BottomNavigationAction
        className="p-1 text-white"
        label="Your Progress"
        value="progress"
        onClick={() => {
          history.push("/");
        }}
        icon={<FontAwesomeIcon className="m-1 text-white" icon={faChartPie} />}
      />
      { validatePassword ? 
      <BottomNavigationAction
        className="p-1 text-white"
        label="More"
        value="signin"
        // onClick={() => {
        //   history.push("/profile");
        // }}
        icon={<FontAwesomeIcon className="m-1 text-white" icon={faEllipsisH} />}
      /> : 
        <BottomNavigationAction
        className="p-1 text-white"
        label="More"
        value="signin"
        onClick={() => {
          history.push("/profile");
        }}
        icon={<FontAwesomeIcon className="m-1 text-white" icon={faEllipsisH} />}
      />
      }
    </BottomNavigation>
  );
};

export default Footer;
