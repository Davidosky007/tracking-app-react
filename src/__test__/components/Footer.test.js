import { BrowserRouter } from 'react-router-dom';
import renderer from 'react-test-renderer';
import Footer from '../../Components/Footer';

jest.mock('react-router-dom', () => ({
  ...jest.requireActual('react-router-dom'),
  useLocation: () => ({
    pathname: 'example.com/test',
  }),
}));

describe('Footer', () => {
  it('renders correctly', () => {
    const tree = renderer.create(
      <BrowserRouter>
        <Footer />
      </BrowserRouter>,
    ).toJSON();
    expect(tree).toMatchSnapshot();
  });
});
