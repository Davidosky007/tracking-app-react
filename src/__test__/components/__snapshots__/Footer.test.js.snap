// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Footer renders correctly 1`] = `
<footer
  className="flex fixed footer"
>
  <a
    className="icon background-dark-gray color-white"
    href="/add"
    onClick={[Function]}
  >
    <div>
      <div
        className="fas fa-plus-square big"
      />
      <div
        className="small m-top-10"
      >
        Add Measure.
      </div>
    </div>
  </a>
  <a
    className="icon background-dark-gray color-white"
    href="/"
    onClick={[Function]}
  >
    <div>
      <div
        className="fas fa-home big"
      />
      <div
        className="small m-top-10"
      >
        Home
      </div>
    </div>
  </a>
  <a
    className="icon background-dark-gray color-white"
    href="/progress"
    onClick={[Function]}
  >
    <div>
      <div
        className="fas fa-weight big"
      />
      <div
        className="small m-top-10"
      >
        Progress
      </div>
    </div>
  </a>
  <a
    className="icon background-dark-gray color-white"
    href="/users/login"
    onClick={[Function]}
  >
    <div>
      <div
        className="fas fa-sign-out-alt big"
      />
      <div
        className="small m-top-10"
      >
        Logout
      </div>
    </div>
  </a>
</footer>
`;
